components/Tabs2/src/MDCTabBarView.m

```diff
@@ -16,4 +16,51 @@
 
 @implementation MDCTabBarView
 
+#pragma mark - Properties
+
+- (void)setItems:(NSArray<UITabBarItem *> *)items {
+  NSAssert([NSThread isMainThread], @"Item array may only be set on the main thread");
+  NSParameterAssert(items != nil);
+
+  if (_items != items && ![_items isEqual:items]) {
+    _items = [items copy];
+
+    // Determine new selected item, defaulting to the first item.
+    UITabBarItem *newSelectedItem = _items.firstObject;
+    if (_selectedItem && [_items containsObject:_selectedItem]) {
+      // Previously-selected item still around: Preserve selection.
+      newSelectedItem = _selectedItem;
+    }
+
+    // Update _selectedItem directly so it's available for -reload.
+    _selectedItem = newSelectedItem;
+  }
+}
+
+- (void)setSelectedItem:(nullable UITabBarItem *)selectedItem {
+  [self setSelectedItem:selectedItem animated:NO];
+}
+
+- (void)setSelectedItem:(nullable UITabBarItem *)selectedItem animated:(BOOL)animated {
+  if (_selectedItem != selectedItem) {
+    NSUInteger itemIndex = [self indexForItem:selectedItem];
+    if (selectedItem && (itemIndex == NSNotFound)) {
+      [[NSException exceptionWithName:NSInvalidArgumentException
+                               reason:@"Invalid item"
```

Comment:
I love the exception raising! ❤️ 

Nit: Please add a slightly more verbose message. Something like, 

```
[NSString stringWithFormat:"(%@) is not a member of the tab bar's `items`.", selectedItem]
```

Feel free to go with something else, but I'd prefer a more descriptive exception.

Reply:
Updated!

Reply:
UITabBar's behavior when the item is not found is to not adjust the `selectedItem`.

I recommend we follow UIKit's behavior rather than introducing a new one.

```objc
UITabBar *tabBar = [[UITabBar alloc] initWithFrame:CGRectZero];
UITabBarItem * item1 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:1];
UITabBarItem * item2 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:2];
UITabBarItem * item3 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:3];
tabBar.items = @[item1, item2 ];

NSLog(@"%@", @(tabBar.selectedItem.tag)); // 0

tabBar.selectedItem = item1;
NSLog(@"%@", @(tabBar.selectedItem.tag)); // 1

tabBar.selectedItem = item2;
NSLog(@"%@", @(tabBar.selectedItem.tag)); // 2

tabBar.selectedItem = item3;
NSLog(@"%@", @(tabBar.selectedItem.tag)); // 2
```

Reply:
This was taken from MDCTabBar, so changing will be an intentional deviation from MDCTabBar.  However, since UIKit just ignores these errors (no messages, no exceptions), I think it's fine if the MDCTabBarView does as well.